server/handler/sessionHelper.go

```diff
@@ -56,20 +59,31 @@ func getSessionHelper() (*SessionHelper, error) {
 	return helperInstance, nil
 }
 
-func (s *SessionHelper) HasSession(r *http.Request) (bool, error) {
+func (s *SessionHelper) ValidSession(r *http.Request) (bool, error) {
 	aSession, err := s.store.Get(r, s.sessionName)
 	if err != nil {
 		return false, err
 	}
+	if aSession.IsNew {
+		return false, nil
+	}
+	// For SSO, verify state
+	if config.IsSsoEnabled() {
```

Comment:
Checking for valid SSO session here means that no API calls can be made until the user has gone through SSO flow.
SSO session should be kept separate.

Reply:
I will resolve this in my PR.